src/model.py

```diff
@@ -403,10 +415,26 @@ async def generate(self, request):
             )
 
             async for output in response_iterator:
-                if response_sender.is_cancelled():
+                is_cancelled = response_state[1]
+                if not stream:
+                    is_cancelled = response_sender.is_cancelled()
+                if is_cancelled:
                     self.logger.log_info("[vllm] Cancelling the request")
                     await self.llm_engine.abort(request_id)
                     self.logger.log_info("[vllm] Successfully cancelled the request")
+                    if stream:
+                        response_state[2] = True  # last response ready to be sent
+                        response = pb_utils.InferenceResponse(
+                            error=pb_utils.TritonError(
```

Comment:
Is this the logic we go by, i.e. if response is cancelled, we still send a response back with the message that it was cancelled?

Reply:
yes, for example: https://github.com/triton-inference-server/python_backend/blob/main/README.md#request-cancellation-handling

Reply:
for decoupled, we also need to send complete final flag

Reply:
thanks! I would recommend changing `user cancelled` to `cancelled on the client side` or something like this, because it might be that request was cancelled by an intermediary system not related to user.

Reply:
sure, updated the message [Update cancellation message](https://github.com/triton-inference-server/vllm_backend/pull/54/commits/0b996e85c011ba26e928410b2039fe7b6bffdf3a)